staging/src/k8s.io/apimachinery/pkg/util/wait/loop_test.go

```diff
@@ -203,45 +204,89 @@ func Test_loopConditionUntilContext_semantic(t *testing.T) {
 			attemptsExpected: 0,
 			errExpected:      context.DeadlineExceeded,
 		},
+		{
+			name:      "context canceled before the second execution and immediate",
+			immediate: true,
+			context: func() (context.Context, context.CancelFunc) {
+				return context.WithTimeout(context.Background(), time.Second)
+			},
+			callback: func(attempts int) (bool, error) {
+				return false, nil
+			},
+			attemptsExpected: 1,
+			errExpected:      context.DeadlineExceeded,
+			timer:            Backoff{Duration: 2 * time.Second}.Timer(),
+		},
+		{
+			name:      "long duration of condition and sliding false",
+			immediate: true,
+			sliding:   false,
+			context: func() (context.Context, context.CancelFunc) {
+				return context.WithTimeout(context.Background(), 5*time.Second)
+			},
+			callback: func(attempts int) (bool, error) {
+				if attempts >= 4 {
+					return true, nil
+				}
+				time.Sleep(time.Second)
+				return false, nil
+			},
+			attemptsExpected: 4,
+			timer:            Backoff{Duration: time.Second, Jitter: 0.001}.Timer(),
+		},
+		{
+			name:      "long duration of condition and sliding true",
+			immediate: true,
+			sliding:   true,
+			context: func() (context.Context, context.CancelFunc) {
+				return context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
5 seconds is very expensive, I think we can have the same functionality this way

```
diff --git a/staging/src/k8s.io/apimachinery/pkg/util/wait/loop_test.go b/staging/src/k8s.io/apimachinery/pkg/util/wait/loop_test.go
index af8f0c9a721..fa984cc0d5a 100644
--- a/staging/src/k8s.io/apimachinery/pkg/util/wait/loop_test.go
+++ b/staging/src/k8s.io/apimachinery/pkg/util/wait/loop_test.go
@@ -218,11 +218,11 @@ func Test_loopConditionUntilContext_semantic(t *testing.T) {
                        timer:            Backoff{Duration: 2 * time.Second}.Timer(),
                },
                {
-                       name:      "long duration of condition and sliding false",
+                       name:      "inmidiate and long duration of condition and sliding false",
                        immediate: true,
                        sliding:   false,
                        context: func() (context.Context, context.CancelFunc) {
-                               return context.WithTimeout(context.Background(), 5*time.Second)
+                               return context.WithTimeout(context.Background(), time.Second)
                        },
                        callback: func(attempts int) (bool, error) {
                                if attempts >= 4 {
@@ -231,15 +231,16 @@ func Test_loopConditionUntilContext_semantic(t *testing.T) {
                                time.Sleep(time.Second)
                                return false, nil
                        },
-                       attemptsExpected: 4,
-                       timer:            Backoff{Duration: time.Second, Jitter: 0.001}.Timer(),
+                       errExpected:      context.DeadlineExceeded,
+                       attemptsExpected: 2,
+                       timer:            Backoff{Duration: 100 * time.Millisecond}.Timer(),
                },
                {
-                       name:      "long duration of condition and sliding true",
+                       name:      "inmidiate and long duration of condition and sliding true",
                        immediate: true,
                        sliding:   true,
                        context: func() (context.Context, context.CancelFunc) {
-                               return context.WithTimeout(context.Background(), 5*time.Second)
+                               return context.WithTimeout(context.Background(), time.Second)
                        },
                        callback: func(attempts int) (bool, error) {
                                if attempts >= 4 {
@@ -249,8 +250,8 @@ func Test_loopConditionUntilContext_semantic(t *testing.T) {
                                return false, nil
                        },
                        errExpected:      context.DeadlineExceeded,
-                       attemptsExpected: 3,
-                       timer:            Backoff{Duration: time.Second, Jitter: 0.001}.Timer(),
+                       attemptsExpected: 1,
+                       timer:            Backoff{Duration: 100 * time.Millisecond}.Timer(),
                },
        }
```

Reply:
5 seconds is indeed expensive, I reduced the duration in each test to 20% of the original, and it seems works fine.
But I don't think we can remove the `Jitter` here, cause if so, we'll create a `fixedTimer` which doesn't implement the `Next` right?
```
func (t *fixedTimer) Next() {
	// no-op for fixed timers
}
```